flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/SchemaEvolveE2eITCase.java

```diff
@@ -0,0 +1,758 @@
+        waitUntilJobRunning(Duration.ofSeconds(30));
+        LOG.info("Pipeline job is running");
+        validateSnapshotData("members");
+
+        LOG.info("Starting schema evolution");
+        String mysqlJdbcUrl =
+                String.format(
+                        "jdbc:mysql://%s:%s/%s",
+                        MYSQL.getHost(),
+                        MYSQL.getDatabasePort(),
+                        schemaEvolveDatabase.getDatabaseName());
+
+        try (Connection conn =
+                        DriverManager.getConnection(
+                                mysqlJdbcUrl, MYSQL_TEST_USER, MYSQL_TEST_PASSWORD);
+                Statement stmt = conn.createStatement()) {
+            waitForIncrementalStage("members", stmt);
+
+            // triggers AddColumnEvent
+            stmt.execute("ALTER TABLE members ADD COLUMN gender TINYINT AFTER age;");
+            stmt.execute("INSERT INTO members VALUES (1012, 'Eve', 17, 0);");
+
+            // triggers AlterColumnTypeEvent and RenameColumnEvent
+            stmt.execute("ALTER TABLE members CHANGE COLUMN age precise_age DOUBLE;");
+
+            // triggers RenameColumnEvent
+            stmt.execute("ALTER TABLE members RENAME COLUMN gender TO biological_sex;");
+
+            // triggers DropColumnEvent
+            stmt.execute("ALTER TABLE members DROP COLUMN biological_sex;");
+            stmt.execute("INSERT INTO members VALUES (1013, 'Fiona', 16);");
+        }
+
+        List<String> expected =
+                Stream.of(
+                                "AddColumnEvent{tableId=%s.members, addedColumns=[ColumnWithPosition{column=`gender` TINYINT, position=AFTER, existedColumnName=age}]}",
+                                "DataChangeEvent{tableId=%s.members, before=[], after=[1012, Eve, 17, 0], op=INSERT, meta=()}",
+                                "AlterColumnTypeEvent{tableId=%s.members, nameMapping={age=DOUBLE}}",
+                                "RenameColumnEvent{tableId=%s.members, nameMapping={age=precise_age}}",
+                                "RenameColumnEvent{tableId=%s.members, nameMapping={gender=biological_sex}}",
+                                "DataChangeEvent{tableId=%s.members, before=[], after=[1013, Fiona, 16.0, null], op=INSERT, meta=()}")
+                        .map(s -> String.format(s, schemaEvolveDatabase.getDatabaseName()))
+                        .collect(Collectors.toList());
+
+        validateResult(expected, taskManagerConsumer);
+
+        waitUntilSpecificEvent(
+                String.format(
+                        "Ignored schema change DropColumnEvent{tableId=%s.members, droppedColumnNames=[biological_sex]} to table %s.members.",
+                        schemaEvolveDatabase.getDatabaseName(),
+                        schemaEvolveDatabase.getDatabaseName()),
+                jobManagerConsumer);
+    }
+
+    @Test
+    public void testLenientSchemaEvolution() throws Exception {
```

Comment:
minor: I like the these tests, but parameterized tests will make the test code shorter